### PR TITLE
Add more types to HttpResponse and StreamingHttpResponse

### PR DESCRIPTION
**Pull request by danifus**
Type hint for `HttpResponse.getvalue()` was missing.

Also adds more specific types for:

- `HttpResponse.content`
- `StreamingHttpResponse.streaming_content`

Not in this PR but as `StreamingHttpResponse.content` just raises `AttributeError` should we exclude it from the type hint for that property?

```
class StreamingHttpResponse(...):

    @property
    def content(self):
        raise AttributeError(
            "This %s instance has no `content` attribute. Use "
            "`streaming_content` instead." % self.__class__.__name__
        )
```

### CODE REVIEW

File: django-stubs/http/response.pyi [hunk 1]
```diff
@@ -62,7 +62,10 @@ class HttpResponseBase(Iterable[Any]):
     def __iter__(self) -> Iterator[Any]: ...
 
 class HttpResponse(HttpResponseBase):
-    content: Any
+    @property
+    def content(self) -> bytes: ...
+    @content.setter
+    def content(self, value: Any): ...
```
Comment (sobolevn) [inline django-stubs/http/response.pyi]: Shouldn't `setter` be annotated as `-> None`?

File: django-stubs/http/response.pyi [hunk 2]
```diff
@@ -83,12 +86,16 @@ class HttpResponse(HttpResponseBase):
...
     def json(self) -> Any: ...
+    def getvalue(self) -> bytes: ...
 
 class StreamingHttpResponse(HttpResponseBase):
     content: Any
-    streaming_content: Iterator[Any]
+    @property
+    def streaming_content(self) -> Iterator[bytes]: ...
+    @streaming_content.setter
+    def streaming_content(self, value: Iterable[Any]): ...
```
Comment (sobolevn) [inline django-stubs/http/response.pyi]: The same here.

**Review by sobolevn — COMMENTED** [review]
> Not in this PR but as StreamingHttpResponse.content just raises AttributeError should we exclude it from the type hint for that property?

I think we can ignore it for now.

Thanks a lot for your PR!

File: django-stubs/http/response.pyi (hunk 1 above)
Comment (danifus) [reply django-stubs/http/response.pyi]: Yeah, they should be there. I've added another commit for that.

**Review by sobolevn — APPROVED** [review]

**Comment by danifus** [comment]
Hi,

Just checking in to see if this was ok to merge? The build failure was due to the pipeline issues you had in June. Is there a way to clear them?

**Comment by sobolevn** [comment]
@danifus yes, can you please rebase the build?

**Comment by danifus** [comment]
Looks like having different types accepted for a setter and returned from a getter is not supported yet:

https://github.com/python/mypy/issues/3004

That thread initially says that it won't be supported but during discussion they decide it is a common idiom and will support it once an approach is agreed upon.

I'll leave the `content` property alone for the time being and create a new pull request for adding the missing `getvalue()` method. Feel free to close this pull request or leave it open for when the mypy fixes arrive.

**Comment by sobolevn** [comment]
@danifus agreed, let's wait for the `mypy` fix! 👍

**Comment by terencehonles** [comment]
@danifus I believe you should be able to close this now that #712 is merged since this is approximately the same change

**Comment by danifus** [comment]
Thanks @terencehonles! Nice work :)

**Pull request closed without merging**